src/schemes/boundary/dummy_particles/dummy_particles.jl

```diff
@@ -324,13 +324,32 @@ function compute_pressure!(boundary_model, ::AdamiPressureExtrapolation,
         v_neighbor_system = wrap_v(v_ode, neighbor_system, semi)
         u_neighbor_system = wrap_u(u_ode, neighbor_system, semi)
 
-        nhs = get_neighborhood_search(system, neighbor_system, semi)
-
         neighbor_coords = current_coordinates(u_neighbor_system, neighbor_system)
 
-        adami_pressure_extrapolation!(boundary_model, system, neighbor_system,
-                                      system_coords, neighbor_coords,
-                                      v_neighbor_system, nhs)
+        # This is an optimization for simulations with large and complex boundaries.
+        # Especially, in 3D simulations with large and/or complex structures outside
+        # of areas with permanent flow.
+        # Note: The version iterating neighbors first is not thread parallizable.
+        # The factor is based on the achievable speed-up of the thread parallizable version.
+        if nparticles(system) >
+           ceil(Int, 0.5 * Threads.nthreads()) * nparticles(neighbor_system)
+            nhs = get_neighborhood_search(neighbor_system, system, semi)
+
+            adami_pressure_extrapolation_neighbor!(boundary_model, system, neighbor_system,
+                                                   system_coords, neighbor_coords,
+                                                   v_neighbor_system, nhs)
+        else
+            nhs = get_neighborhood_search(system, neighbor_system, semi)
+
+            adami_pressure_extrapolation!(boundary_model, system, neighbor_system,
+                                          system_coords, neighbor_coords,
+                                          v_neighbor_system, nhs)
+        end
+        @simd for particle in eachparticle(system)
```

Comment:
```suggestion
        for particle in eachparticle(system)
```
There is absolutely no difference for me:
```
julia> @btime f($boundary_system, $(boundary_system.boundary_model.pressure))
  12.708 μs (0 allocations: 0 bytes)

julia> @btime f_inbounds($boundary_system, $(boundary_system.boundary_model.pressure))
  12.708 μs (0 allocations: 0 bytes)

julia> @btime f_simd($boundary_system, $(boundary_system.boundary_model.pressure))
  12.708 μs (0 allocations: 0 bytes)
```
Did you see a speedup on your machine?

Reply:
I did not try this.